openlibrary/plugins/upstream/models.py

```diff
@@ -691,48 +698,118 @@ def get_sorted_editions(
-                    edition_keys += web.ctx.site.things(query)
-            else:
-                db_query["ocaid~"] = "*"
+        def get_solr_editions():
+            work = self._solr_data
+            edition_keys = work.get('edition_key') or []
+            edition_queries = [f'key:"/books/{key}"' for key in edition_keys]
+
+            ed_results = run_solr_query(
+                EditionSearchScheme(),
+                {'q': ' OR '.join(edition_queries)},
+                rows=limit,
+                fields=[
+                    'ia',
+                    'key',
+                    'title',
+                    'subtitle',
+                    'language',
+                    'isbn',
+                    'publish_date',
+                    'publisher',
+                ],
+                user_query=False,
+            )
 
-        if not edition_keys:
-            solr_is_up_to_date = (
-                self._solr_data
-                and self._solr_data.get('edition_key')
-                and len(self._solr_data.get('edition_key')) == self.edition_count
+            _editions = []
+            if ed_results.docs:
+                for solr_edition in ed_results.docs:
+                    ed = web.storify(solr_edition)
+                    if ed.get('ia', ''):
+                        ed.ocaid = ed.ia
+                    else:
+                        ed.ocaid = ''
+
+                    ed.languages = (
+                        [get_language(f'/languages/{ed.language}')]
+                        if 'language' in ed
+                        else []
+                    )
+                    # XXX : Actually set these, if possible:
+                    ed.oclc_numbers = None
+                    if ed.get('isbn', ''):
+                        # XXX : use `isbn.get_isbn_10_and_13` to assist setting isbns
+                        if len(ed.isbn) == 10:
+                            ed.isbn_10 = [ed.isbn]
+                            ed.isbn_13 = [isbn_10_to_isbn_13(ed.isbn)]
+                        else:
+                            # XXX : Assumption: any ISBN will be an ISBN 10 or ISBN 13
+                            ed.isbn_13 = [ed.isbn]
+                            ed.isbn_10 = [isbn_13_to_isbn_10(ed.isbn)]
+
+                    ed.isbn_10 = ed.isbn_10 if 'isbn_10' in ed else []
+                    ed.isbn_13 = ed.isbn_13 if 'isbn_13' in ed else []
+                    if 'subtitle' not in ed:
+                        ed.subtitle = 'default subtitle'
+                    ed.publishers = [ed.publisher] if ed.publisher else []
+                    if 'publish_date' not in ed:
+                        ed.publish_date = 'default publish_date'
+                    ed.physical_format = 'placeholder physical format'
+                    ed.edition_name = 'placeholder edition name'
+                    # XXX : Use cover_i from solr edition to set cover...
+                    ed.type = 'solr_edition'
```

Comment:
I'm not as keen on this, since `type` is already a property that exists on both Infogami records and solr records